spec/system/category_topics_spec.rb

```diff
@@ -32,3 +32,24 @@
     expect(category_list).to have_no_new_posts_badge
   end
 end
+
+describe "Viewing category topic list", type: :system do
+  let(:topic_list) { PageObjects::Components::TopicList.new }
+
+  context "when parent category has default_list_filter=none" do
```

Comment:
This spec is not related to the fix. But it is adjacent, and I noticed that we had no spec for the `default_list_filter` behaviour.